docs/site/tutorials/authentication/Quick-Authentication-Tutorial-Session-Tokens.md

```diff
@@ -0,0 +1,297 @@
+---
+lang: en
+title: 'Quick introduction to authentication with session tokens'
+keywords: LoopBack 4.0, LoopBack 4, Authentication, Tutorial
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Quick-Authentication-Tutorial-Session-Tokens.html
+summary: A LoopBack 4 application that uses session tokens for authentication
+---
+
+## Overview
+
+Loopbck 4 authentication is explained in detail
+[here](../../Loopback-component-authentication.md) and there is a complete
+tutorial for authentication with JWTs [here](Authentication-Tutorial.md).
+
+But this tutorial summarises just the very basic steps to help you get started
+with simple authentication.
+
+You can adapt the code below to suit your needs, but let's start by imagining
+that we will build an OAuth2-like approach using session tokens:
+
+1. The user will make a POST request to `/user/login` passing a **Basic**
+   authentication header.
+
+   If successful, this will return a session token to the user.
+
+2. On all future requests, the user will present the token via a **Bearer**
+   authentication header.
+
+## Install dependencies
+
+First, ensure you have the necessary packages installed:
+
+```sh
+npm install @loopback/authentication @loopback/security
+```
+
+## Add authentication to the sequence
+
+Authentication is not enabled by default. We need to tell loopback that every
+incoming HTTP request should check to see if authentication is required.
+
+Open `src/sequence.ts` and add the following:
+
+```ts
+// At the top
+
+import { AuthenticationBindings, AuthenticateFn } from '@loopback/authentication';
+
+// In the constructor's arguments
+
+    @inject(AuthenticationBindings.AUTH_ACTION) protected authenticateRequest: AuthenticateFn,
+
+// In the handle function
+
+    const route = this.findRoute(request);
+    await this.authenticateRequest(request);    // ADD THIS LINE
+    const args = await this.parseParams(request, route);
+```
+
+## Create a Basic authentication strategy
+
+The AuthenticationStrategy is where the work really happens. It will extract the
+authentication details from the HTTP request, and check them against the
+database.
+
+Make a new folder `src/strategies`
+
+Create the file `src/strategies/basic-strategy.ts` with the following content:
+
+```ts
+import {AuthenticationStrategy} from '@loopback/authentication';
+import {UserProfile, securityId} from '@loopback/security';
+import {inject} from '@loopback/core';
+import {repository} from '@loopback/repository';
+import {HttpErrors, Request} from '@loopback/rest';
+import {UserRepository} from '../repositories';
+
+export interface Credentials {
+  clientKey: string;
+  clientSecret: string;
+}
+
+export class BasicAuthenticationStrategy implements AuthenticationStrategy {
+  name = 'BasicStrategy';
+
+  constructor() {}
+  //@repository(UserRepository) public userRepository: UserRepository,
```

Comment:
nitpick: this line should be added as the parameter for the constructor instead of outside of the constructor.

Reply:
I really wanted to do that too. But prettier wouldn't let me. It's even strict about TS in Markdown!